06_neighborhoods/Transportation/transit_trips_cost_county_subgroups.R

```diff
@@ -0,0 +1,740 @@
+# B02001_007 Some other race alone
+# B02001_008 Two or more races
+
+
+my_states <- fips_codes %>% 
+  filter(!state %in% c("PR", "UM", "VI", "GU", "AS", "MP")) %>%
+  pull(state) %>%
+  unique()
+
+acs_tract_pop15 <- map_dfr(
+  my_states,
+  ~ get_acs(
+    geography = "tract",
+    state = .,
+    table = "B02001",
+    year = 2015,
+    survey = "acs5",
+    output = "wide"
+  )
+)
+
+# rename columns for clarity
+acs_tract_pop15 <- acs_tract_pop15 %>%  
+  rename(total_population = B02001_001E,
+         white_population = B02001_002E,
+         black_population = B02001_003E,
+         asian_population = B02001_005E,
+         aian_population = B02001_004E,
+         nhpi_population = B02001_006E,
+         other_population = B02001_007E
+  )
+
+acs_tract_pop19 <- map_dfr(
+  my_states,
+  ~ get_acs(
+    geography = "tract",
+    state = .,
+    table = "B02001",
+    year = 2019,
+    survey = "acs5",
+    output = "wide"
+  )
+)
+
+acs_tract_pop19 <- acs_tract_pop19 %>%  
+  rename(total_population = B02001_001E,
+         white_population = B02001_002E,
+         black_population = B02001_003E,
+         asian_population = B02001_005E,
+         aian_population = B02001_004E,
+         nhpi_population = B02001_006E,
+         other_population = B02001_007E
+  )
+
+# keep only vars we want
+acs_tract_pop15 <- acs_tract_pop15 %>%
+  select(GEOID, total_population, white_population, black_population,
+         asian_population, aian_population, nhpi_population, other_population)
+acs_tract_pop19 <- acs_tract_pop19 %>%
+  select(GEOID, total_population, white_population, black_population,
+         asian_population, aian_population, nhpi_population, other_population)
+
+# note that this is 73,056 observations in the ACS data
+# That means we will have 317 missing tract values in 2015 (73056 - 72739), and 
+# 314 missing tract values in the 2019 data (72056 - 72742)
+
+# Join the transit data with the ACS population data,
+
+# first, prep GEOID in the transit data for join purposes
+transit_cost_tracts_2015$GEOID <- paste0(transit_cost_tracts_2015$state, transit_cost_tracts_2015$county, transit_cost_tracts_2015$tract)
+transit_cost_tracts_2019$GEOID <- paste0(transit_cost_tracts_2019$state, transit_cost_tracts_2019$county, transit_cost_tracts_2019$tract)
+transit_trips_tracts_2015$GEOID <- paste0(transit_trips_tracts_2015$state, transit_trips_tracts_2015$county, transit_trips_tracts_2015$tract)
+transit_trips_tracts_2019$GEOID <- paste0(transit_trips_tracts_2019$state, transit_trips_tracts_2019$county, transit_trips_tracts_2019$tract)
+
+
+# merge population data with transit data files (4 of them)
+############################################################################
+# FIRST ONE - Cost in 2015
+############################################################################
+cost_data_2015 <- left_join(acs_tract_pop15, transit_cost_tracts_2015, by = "GEOID")
+#Test with anti_join to make sure it worked properly
+stopifnot(
```

Comment:
These `stopifnot` sections are preventing the script from running all the way through top to bottom. They are useful for testing / debugging but I believe we are supposed to have final code that runs with no errors or issues. So we need to either switch this to some kind of warning output or remove it since we trust the code is working as intended. 

Same comment applies to the other scripts where  `stopifnot` is used.

Reply:
That makes sense -- I've commented these out of the scripts since they are process code not necessary to the output, but am leaving them in as comments to show documentation, missing counts, etc.

Reply:
There are still places where the code is active. Please double check that scripts run through start to finish. Thanks!

Reply:
ahhh!! Sorry!! Removed  & updated